# Welcome to our Combined Notebook!

In this project, we're analyzing the BEST neighborhood to commit crime. We'll be using Police Blotter, Wealth, and Non-Traffic Citation Data to find the most crime-friendly neighborhood!

First, let's import some useful libraries.

In [7]:
# Combined Notebook

import pandas as pd
import numpy as np
import operator

%matplotlib inline

import matplotlib.pyplot as plt

pd.set_option('display.min_rows', 400) # So we see all the neighborhoods
pd.set_option('display.max_rows', 400)

# Zach's Data: Police Blotter Data

First up is the police blotter data, which contains crime statistics dating back to 2005, sorted by type of crime, location, time committed, and more!

Let's display the number of offenses per neighborhood.

In [18]:
# Zach's Data
policeData = pd.read_csv("archive-police-blotter.csv", parse_dates=True, low_memory=False) # Read File

# Drop unnecessary columns
policeData.drop(columns = ["PK", "CCR", "INCIDENTTIME", "INCIDENTLOCATION", "CLEAREDFLAG", "INCIDENTZONE", "HIERARCHYDESC", "INCIDENTTRACT", "X", "Y"], inplace = True)

#Deletes from 8-29 (unnecessary hierarchies)
for i in range(22):
    policeData.drop(policeData[policeData.HIERARCHY == i+8].index, inplace=True)

#Deletes the ones the for loop didn't get to
policeData.drop(policeData[policeData.HIERARCHY == 0].index, inplace=True)
policeData.drop(policeData[policeData.HIERARCHY == 1].index, inplace=True)
policeData.drop(policeData[policeData.HIERARCHY == 2].index, inplace=True)
policeData.drop(policeData[policeData.HIERARCHY == 4].index, inplace=True)
policeData.drop(policeData[policeData.HIERARCHY == 99].index, inplace=True)

# Group by neighborhood
neighborhoods = policeData.groupby(["INCIDENTNEIGHBORHOOD"]).count()

# Remove hierarchy so we can combine all forms of stealing
neighborhoods.drop(columns = "HIERARCHY", inplace = True)

neighborhoods.sort_values("OFFENSES") # Sort values by # of offenses



OFFENSES
INCIDENTNEIGHBORHOOD                 
Outside County                     10
Outside State                      21
Mt. Oliver Boro                    59
Chartiers City                     97
New Homestead                     120
Arlington Heights                 202
Mt. Oliver Neighborhood           208
Glen Hazel                        211
Hays                              211
Summer Hill                       216
Swisshelm Park                    217
West End                          223
East Carnegie                     227
Esplen                            252
Fairywood                         267
Oakwood                           283
Ridgemont                         307
Windgap                           370
Bon Air                           386
Polish Hill                       395
Spring Garden                     422
Regent Square                     463
Chateau                           501
California-Kirkbride              501
Duquesne Heights                  518
Outside City                      527
Fineview                          543
St. Clair                         582
Bedford Dwellings                 622
Morningside                       624
South Shore                       680
Banksville                        694
Allegheny West                    716
Beltzhoover                       747
Spring Hill-City View             748
Stanton Heights                   765
Lincoln Place                     769
Manchester                        787
Upper Hill                        828
Elliott                           860
Overbrook                         896
Crafton Heights                   930
Terrace Village                   951
Northview Heights                 953
Arlington                         969
Westwood                          977
Upper Lawrenceville               982
South Oakland                    1023
Homewood West                    1076
Bluff                            1174
Troy Hill-Herrs Island           1177
Friendship                       1295
West Oakland                     1323
Allegheny Center                 1337
Crawford-Roberts                 1349
North Shore                      1362
Point Breeze North               1376
Middle Hill                      1378
Garfield                         1410
Greenfield                       1443
Perry South                      1455
Lower Lawrenceville              1521
Perry North                      1560
Strip District                   1682
Beechview                        1738
South Side Slopes                1788
East Hills                       1794
Hazelwood                        1848
Central Northside                1868
Point Breeze                     1915
Sheraden                         1989
Marshall-Shadeland               2023
Larimer                          2070
East Allegheny                   2137
Allentown                        2264
Squirrel Hill North              2445
Central Lawrenceville            2500
Brookline                        2513
Homewood North                   2520
Highland Park                    2708
Brighton Heights                 2845
Knoxville                        3025
Homewood South                   3025
Lincoln-Lemington-Belmar         3534
North Oakland                    3603
Mount Washington                 3655
Squirrel Hill South              4012
Central Oakland                  4035
East Liberty                     4094
Carrick                          5317
Bloomfield                       6145
Shadyside                        6770
South Side Flats                 8855
Golden Triangle/Civic Arena     11113

In [12]:
# Renhan's Data

# inDict Function
def inDict(word):
    for w in dictionary:
        if word == w:
            return True
    return False

# Initialize a variable that holds the data
non_traffic_citations = pd.read_csv("NonTrafficCitations.csv") # index_col="_id")

# Filter the columns so that only the column with the neighborhood are saved
neighborhoods = non_traffic_citations.iloc[:, 8]

# Initialize a global dictionary
global dictionary
dictionary = {}

# Iterate this code for each row in neighborhoods
for row in neighborhoods:
    # If the item is in the dictionary, add 1 to its value
    if(inDict(row)):
        dictionary[row] += 1
        
    # If the item is not in the dictionary, create a new entry and initialize it to 1
    else:
        dictionary[row] = 1
        
# Sort the dictionary (this will turn it into a list of tuples)
dictionary = sorted(dictionary.items(), key=operator.itemgetter(1))

# Print the list
for key in dictionary:
    print(str(key[0]) + ": " + str(key[1]))

nan: 1
Ridgemont: 1
Mt. Oliver Boro: 1
Chartiers City: 1
Outside State: 2
Outside County: 2
St. Clair: 3
Mt. Oliver Neighborhood: 4
New Homestead: 5
Swisshelm Park: 5
Spring Garden: 7
Hays: 7
Regent Square: 7
Summer Hill: 7
Arlington Heights: 7
Unable To Retrieve Address: 7
Oakwood: 8
Fairywood: 9
Outside City: 9
Duquesne Heights: 13
Friendship: 14
Windgap: 14
Allegheny West: 16
East Carnegie: 17
Polish Hill: 17
Lower Lawrenceville: 19
Glen Hazel: 19
Morningside: 22
Bon Air: 22
Upper Hill: 23
Esplen: 25
California-Kirkbride: 26
Westwood: 26
Terrace Village: 26
West End: 27
Stanton Heights: 27
Mount Oliver: 28
Upper Lawrenceville: 31
Fineview: 31
Point Breeze North: 32
Manchester: 33
Banksville: 36
Homewood West: 38
Lincoln Place: 38
Troy Hill: 38
Perry North: 39
Golden Triangle/Civic Arena: 42
Highland Park: 42
Spring Hill-City View: 45
Overbrook: 46
South Shore: 46
Point Breeze: 47
Bedford Dwellings: 51
Crafton Heights: 54
Northview Heights: 55
West Oakland: 56
Beltzhoover: 56
Chateau

# Rafael's Data: Total Income Estimate

Lastly we have the total estimated annual income for each neighborhood. This data will be sorted by greatest to least total estimated income for each neighborhood.

In [17]:
# Rafael's Data

# Data (uploading, finding total estimate data, sorting)
wages = pd.read_csv("wage-or-salary-income-in-the-past-12-months-for-households.csv")
totalEstimates = wages.iloc[:, [0,2,3]]
greatestEstimates = totalEstimates.sort_values(by=['Estimate; Total:'], ascending = False)
greatestEstimates

Neighborhood  Estimate; Total:  Margin of Error; Total:
68                  Shadyside            7484.0               360.919659
77        Squirrel Hill South            7211.0               332.601864
13                  Brookline            5843.0               327.694980
9                  Bloomfield            4571.0               205.494525
53           Mount Washington            4385.0               249.523546
15                    Carrick            4301.0               277.613040
7                   Beechview            3567.0               258.027130
36                 Greenfield            3438.0               251.708164
76        Squirrel Hill North            3370.0               187.890926
72           South Side Flats            3311.0               264.567572
12           Brighton Heights            3222.0               192.234232
39              Highland Park            2977.0               243.973359
56              North Oakland            2972.0               230.516811
28               East Liberty            2902.0               249.248872
63               Point Breeze            2342.0               137.433620
17      Central Lawrenceville            2338.0               171.572142
69                   Sheraden            2202.0               195.269045
79            Stanton Heights            2056.0               129.046503
73          South Side Slopes            1988.0               173.329167
16  Central Business District            1968.0               221.438479
46   Lincoln-Lemington-Belmar            1959.0               154.466825
5                  Banksville            1946.0               229.222599
19            Central Oakland            1903.0               157.397586
38                  Hazelwood            1870.0               201.881153
22            Crafton Heights            1677.0               126.795110
62                Perry South            1630.0               164.289379
61                Perry North            1625.0               132.238421
60                  Overbrook            1594.0               104.144131
34                   Garfield            1571.0               116.914499
49         Marshall-Shadeland            1565.0               182.655413
51                Morningside            1534.0               141.555643
43                  Knoxville            1510.0               218.716712
89                   Westwood            1492.0               112.538882
18          Central Northside            1479.0               113.929803
52       Mount Oliver Borough            1465.0               173.890770
45              Lincoln Place            1460.0               139.907112
27                 East Hills            1325.0               165.508308
25             East Allegheny            1302.0               155.051604
86        Upper Lawrenceville            1287.0               137.091211
47        Lower Lawrenceville            1257.0               125.399362
40             Homewood North            1210.0               129.645671
24           Duquesne Heights            1170.0               125.207827
84                  Troy Hill            1158.0               147.800541
70              South Oakland            1155.0               150.512458
23           Crawford-Roberts            1125.0               138.592929
41             Homewood South            1082.0               116.901668
75      Spring Hill-City View            1080.0               138.166566
33                 Friendship            1074.0               122.065556
2                   Allentown            1069.0               132.457540
64         Point Breeze North             996.0               148.721216
48                 Manchester             984.0               119.331471
29                    Elliott             953.0               120.370262
85                 Upper Hill             942.0               159.965621
50                Middle Hill             926.0               122.086035
0 

# Data Analysis

For our project, we decided to grade based on two types of criminals: those that are confident in their skills, and those that are relatively new and inexperienced.

For the first type of criminal, we'll assume that the best neighborhoods to rob are those with a low crime rate; this means people won't be expecting crime as much. Obviously, there is the possibility that crime is low because the police are really good or because security is tight, but we'll assume that a good criminal will not care or will be able to bypass these obstacles with skills. This person would have to be a CRIMINAL MASTERMIND, as depicted below.

![Criminal Mastermind](https://cdn.discordapp.com/attachments/272551180415270912/963926885425872928/unknown.png)


For the second type of criminal, I'm going to assume they're inexperienced, and would like to commit crime in a neighborhood that is easier to succeed without being caught. This would be a neighborhood with a high crime rate. See picture below for more information.

![Robber](https://cdn.discordapp.com/attachments/272551180415270912/963925109029085245/unknown.png)

For the high-class criminal, wealthy, low crime/citation rate neighborhoods would be best.
For the low-class criminal, somewhat wealthy, high crime/citation rate neighborhoods would be best.

For the Police Blotter Data, the neighborhoods with the highest # of offenses were Golden Triangle, South Side Flats, and Shadyside, each with over 6000 offenses. The neighborhoods with the lowest # of offenses were Mt. Oliver Borough, Chartiers City, and New Homestead, all of which have less than 200 offenses.

For the Non-Traffic Citations, the neighborhoods with the highest # of citations were Central Oakland, Central Business District, and South Side Flats. The neighborhoods with the lowest # of citations were Ridgemont, Mount Oliver Borough, and Chartiers City.

The wealthiest neighborhoods are Shadyside, Squirrel Hill South, Brookline, Bloomfield, and Mount Washington.

For our metric, we will rate each of these categories out of ten and combine them (except wealth, which we will rate out of 5, considering it an "added bonus" for the criminal if they are wealthy). The neighborhood(s) with the best scores for the two types of criminals will be the best. Rather than rating each neighborhood (as there are WAY too many), we'll take the top 4 for Police Blotter/Non-Traffic Incidents for each type of criminal and rate those.

## First Type of Criminal

### Mt. Oliver Borough: 
    Police Blotter Incidents: 59 offenses, 10/10
    Non-Citation Incidents: 1 citation, 10/10
    Wealth: 1/5
    
    Total: 21/25
### Chartiers City:
    Police Blotter Incidents: 97 offenses, 9.5/10
    Non-Citation Incidents: 1 citation, 10/10
    Wealth: 1/5
    
    Total: 20.5/25
### New Homestead:
    Police Blotter Incidents: 120 offenses, 9/10
    Non-Citation Incidents: 5 citation, 9/10
    Wealth: 1.25/5
    
    Total: 19.25/25
### Ridgemont:
    Police Blotter Incidents: 307 offenses, 7.5/10
    Non-Citation Incidents: 1 citation, 10/10
    Wealth: 1.5/5
    
    Total: 19/25
    
## Second Type of Criminal

### Golden Triangle:
    Police Blotter Incidents: 11113 offenses, 10/10
    Non-Citation Incidents: 4/10
    Wealth: N/A
    
    Total: 14/20
### South Side Flats: 
    Police Blotter Incidents: 8855 offenses, 9.5/10
    Non-Citation Incidents: 3082 incidents, 10/10
    Wealth: 4.25/5
    
    Total: 23.75/25
### Shady Side: 
    Police Blotter Incidents: 6770 offenses, 8/10
    Non-Citation Incidents: 285 incidents, 7/10
    Wealth: 5/5
    
    Total: 20/25
### Central Oakland: 
    Police Blotter Incidents: 4035 offenses, 6.5/10
    Non-Citation Incidents: 622 incidents, 8/10
    Wealth: 4/5
    
    Total: 18.5/25
    
# Conclusions

To conclude, for the high-class (first type) of criminal, Mount Oliver Borough is the best neighborhood! It has the lowest crime rate, although its wealth rating is not very high. Despite that, there should be at least a few houses with some valuable loot waiting to be plundered.

For the low-class (second type) of criminal, South Side Flats would be the best. It is very wealthy, and has an incredibly high number of incidents. It is the perfect neighborhood.